x-pack/libbeat/management/tests/mbtest/metricbeat_v2_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package mbtest
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	// initialize the plugin system before libbeat does, so we can overwrite it properly
+	_ "github.com/elastic/beats/v7/x-pack/libbeat/management"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management/tests"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/cmd"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+)
+
+var expectedMBStreams = &proto.UnitExpectedConfig{
+	DataStream: &proto.DataStream{
+		Namespace: "default",
+	},
+	Type:     "system/metrics",
+	Id:       "system/metrics-system-default-system",
+	Name:     "system-1",
+	Revision: 1,
+	Meta: &proto.Meta{
+		Package: &proto.Package{
+			Name:    "system",
+			Version: "1.17.0",
+		},
+	},
+}
+
+func TestSingleMetricbeatMetricsetWithProcessors(t *testing.T) {
+	tests.InitBeatsForTest(t, cmd.RootCmd)
+	var mbStreams = []*proto.Stream{
+		{
+			Id: "system/metrics-system.cpu-default-system",
+			DataStream: &proto.DataStream{
+				Dataset: "system.cpu",
+				Type:    "metrics",
+			},
+			Source: tests.RequireNewStruct(map[string]interface{}{
+				"metricsets": []interface{}{"cpu"},
```

Comment:
The same comment I made for Filebeat applies here. Is this as complex as a configuration can get? How well is this testing the transformations?

We can at least include multiple metricsets.

Reply:
90% of the transformations are the `add_field` processors, which these tests cover pretty well.

As far as other modules, I mostly went with `system` since I knew those would be available across most environments, I wasn't sure if we wanted to get into a situation where we're also spinning up a redis container or whatever just for the sake of one integration test.